### PR TITLE
fix: minor cost-tracking review findings

### PR DESCRIPTION
**Pull request by AnaBerg**
Applies the MINOR findings from the AIEDEV-1 (Factory Stats v2) review of ff99597..3eb5514, re-evaluated against main after #506 (per-run usage semantics) merged.

Part of [AIEDEV-1](https://linear.app/nimble-la/issue/AIEDEV-1)

## Items

1. **hub_pricing must not override a known gateway cost** — verified already subsumed by the #506 guard (`!(oldSource == "gateway" && source == "hub_pricing")` + restore path); added a regression test pinning that a cost-less heartbeat leaves cost, `cost_source` and `committed_cost_usd` untouched. (594116f)
2. **usage_daily clamp could swallow other sessions' spend** — corrections now land on the day the run's cost was originally recorded (new `task_run_usage.usage_day` column, backfilled on migration from `updated_at`), so the `MAX(0, ...)` clamp is removed; a test proves one session's cross-day correction can't erase another session's spend. (bce09ac, 8db0a2c)
3. **Model attribution on empty/changed model** — an empty-model heartbeat no longer clobbers the stored model; each run's tokens are attributed to the model reported with that run. Folded into bce09ac (same lines as item 2's change; splitting would have left an inconsistent intermediate commit).
4. **Projection nits** — projection now excludes today's partial spend and projects today as a full mean day (`remaining = daysInMonth - day + 1`); confidence uses complete days (`day - 1`); CoV uses sample std dev (n-1). Tests updated, incl. boundary cases at 7/14 complete days. (3b37462)
5. **\$0.00 vs no-data ambiguity** — usage fields on the runs API are now pointers driven by `usage_updated_at`: recorded usage emits explicit values (including zeros), never-recorded runs omit the fields. Frontend needed no changes — `web/lib/types.ts` and the "—"/drill-down logic already distinguish `undefined` from `0` correctly, so no UI screenshots (rendering change only manifests for recorded-zero runs). (4b725ee)
6. **Static pricing coverage** — seeded the gpt-5 family (gpt-5, gpt-5-mini, gpt-5-nano, gpt-5.1, gpt-5.6) at current list prices with OpenAI-style cache pricing; cost-less heartbeats with no price match are logged once per model. (1f1318a)

Also fixes in passing (4b725ee): `pkg/hub/task_run_analytics_api_test.go:249` referenced the removed `TicketToPr.Authoritative` field, so `go test ./pkg/hub` did not compile on main.

## Review & tests

- Implemented via three parallel gpt-5.6 (Codex) streams; independently reviewed by opus-4.8 and gpt-5.6. The single converged finding (missing `usage_day` backfill for pre-migration rows) is fixed in 8db0a2c.
- `go build ./...` OK; `go test ./pkg/hub -count=1` OK (17s, includes 7 new/updated usage tests + migration backfill test); `cd web && npm run build` OK (TypeScript check passed, 34 static pages).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/hub/task_run_usage.go [hunk 1]
```diff
@@ -84,16 +97,21 @@ func (s *Server) recordTaskRunUsage(clawID string, snapshot taskRunUsageSnapshot
...
-	_, err = tx.Exec(`INSERT INTO usage_daily(tenant_id,day,workspace_name,factory_name,workflow_name,model,input_tokens,output_tokens,total_tokens,cost_usd,updated_at) VALUES(?,?,?,?,?,?,?,?,?,MAX(0,?),?) ON CONFLICT(tenant_id,day,workspace_name,factory_name,workflow_name,model) DO UPDATE SET input_tokens=input_tokens+excluded.input_tokens,output_tokens=output_tokens+excluded.output_tokens,total_tokens=total_tokens+excluded.total_tokens,cost_usd=MAX(0,cost_usd+?),updated_at=excluded.updated_at`, tenant, day, workspace, factory, workflow, snapshot.Model, din, dout, dtotal, dcost, ts, dcost)
+	_, err = tx.Exec(`INSERT INTO usage_daily(tenant_id,day,workspace_name,factory_name,workflow_name,model,input_tokens,output_tokens,total_tokens,cost_usd,updated_at) VALUES(?,?,?,?,?,?,?,?,?,?,?) ON CONFLICT(tenant_id,day,workspace_name,factory_name,workflow_name,model) DO UPDATE SET input_tokens=input_tokens+excluded.input_tokens,output_tokens=output_tokens+excluded.output_tokens,total_tokens=total_tokens+excluded.total_tokens,cost_usd=cost_usd+excluded.cost_usd,updated_at=excluded.updated_at`, tenant, usageDay, workspace, factory, workflow, effectiveModel, din, dout, dtotal, dcost, ts)
```
[the remark this answers is not in the document]
Comment (AnaBerg) [reply pkg/hub/task_run_usage.go:114]: Fixed in 109f86e: cost-only heartbeats now keep the stored model for both the row and the usage_daily bucket (model changes take effect at the next run), with a regression test proving the correction lands in the original model's bucket.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: pkg/hub/task_run_usage.go [hunk 2]
```diff
@@ -84,16 +97,21 @@ func (s *Server) recordTaskRunUsage(clawID string, snapshot taskRunUsageSnapshot
 		cost = oldCost
 		source = oldSource
 	}
-	ts := now().UnixMilli()
-	_, err = tx.Exec(`INSERT INTO task_run_usage(id,tenant_id,run_id,session_key,model,model_provider,input_tokens,output_tokens,total_tokens,committed_input_tokens,committed_output_tokens,committed_total_tokens,committed_cost_usd,estimated_cost_usd,cost_source,first_seen_at,updated_at) VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?) ON CONFLICT(tenant_id,run_id,session_key) DO UPDATE SET model=excluded.model,model_provider=excluded.model_provider,input_tokens=excluded.input_tokens,output_tokens=excluded.output_tokens,total_tokens=excluded.total_tokens,committed_input_tokens=excluded.committed_input_tokens,committed_output_tokens=excluded.committed_output_tokens,committed_total_tokens=excluded.committed_total_tokens,committed_cost_usd=excluded.committed_cost_usd,estimated_cost_usd=excluded.estimated_cost_usd,cost_source=excluded.cost_source,updated_at=excluded.updated_at`, uuid.NewString(), tenant, runID, snapshot.SessionKey, snapshot.Model, snapshot.ModelProvider, in, out, total, comIn, comOut, comTotal, comCost, nullFloat(cost), source, ts, ts)
+	usageNow := now()
+	if s.nowFunc != nil {
+		usageNow = s.nowFunc()
+	}
```
[the remark this answers is not in the document]
Comment (AnaBerg) [reply pkg/hub/task_run_usage.go]: Fixed in 109f86e: replaced the redundant now() read with the existing s.reaperNow() helper (already used outside the reaper in doctor_runtime.go and pipeline_runner.go).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
